### ISSUE TITLE
Snackbar: Focus restoration when not part of `SnackbarList`

### ISSUE THREAD

**Issue by mirka** [WordPress/gutenberg#38425]
## What problem does this address?

As surfaced in #38365, the [focus restoration logic after dismissing a `Snackbar`](https://github.com/WordPress/gutenberg/blob/d257d326075fafeb473adcf16c7ed1d2b54ac384/packages/components/src/snackbar/index.js#L67) only works when it is rendered inside a `SnackbarList`.

### Possible solution 1: Return to the previous tabbable

Use [`tabbable.findPrevious()`](https://github.com/WordPress/gutenberg/blob/d257d326075fafeb473adcf16c7ed1d2b54ac384/packages/dom/src/tabbable.js#L164). This could perhaps replace the `listRef` logic altogether.

### Possible solution 2: Enforce usage of `SnackbarList`

It may be that we actually do not want consumers to render `Snackbar` outside of `SnackbarList`. In this case, this guideline should at least be documented better.

**Comment by ciampo** [comment]
We could maybe go with both? Having a way to discourage using `Snackbar` outside of `SnackbarList`, and returning to previous tabbable as a fallback in case that happens?

> It may be that we actually do not want consumers to render `Snackbar` outside of `SnackbarList`. In this case, this guideline should at least be documented better.

It would be interesting to see what would be the best way to implement this. We do have other components that are also meant to be used in combination with each other (e.g. `ItemGroup` and `Item`, `ToolsPanel` and `ToolsPanelItem`, ...)
